gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/ZipModule.kt

```diff
@@ -46,10 +48,45 @@ open class ZipModule @Inject constructor(objects: ObjectFactory) : DefaultTask()
         val unsignedFile = unsignedModule.get()
         val contentDir = content.get().asFile
 
+        checkDuplicateJars(contentDir)
+
         project.logger.info("Zipping '${contentDir.absolutePath}' into ' ${unsignedFile.asFile.absolutePath}'")
         project.ant.invokeMethod(
             "zip",
             mapOf("basedir" to contentDir, "destfile" to unsignedFile)
         )
     }
+
+    /**
+     * Try to detect jars in contentDir with the same name but have multiple
+     * versions existed.  Fail the build when it's found.
+     **/
+    fun checkDuplicateJars(contentDir: File) {
+        project.logger.info("Parsing file name in: ${contentDir.absolutePath}")
+
+        val fileMap = mutableMapOf<String, String>()
+        val regex = "^(.+?)-(\\d+.*)(?:\\.jar)".toRegex()
+
+        contentDir.walk().filter { it.isFile }.forEach { file ->
+            val matchResult = regex.find(file.name) // Match all the jar files
+
+            if (matchResult != null) {
+                val name = matchResult.groupValues[1]
+                val version = matchResult.groupValues[2]
+
+                if (fileMap.containsKey(name)) {
+                    throw TaskExecutionException(
+                        this,
+                        IllegalArgumentException(
```

Comment:
Probably overkill to generate a nested exception. The message is going to be enough context for an un-nested exception of either type.